compiler-core/src/ast.rs

```diff
@@ -467,13 +470,15 @@ pub struct Function<T, Expr> {
     pub location: SrcSpan,
     pub end_position: u32,
     pub name: EcoString,
+    pub name_location: Option<SrcSpan>,
```

Comment:
How come the name is optional?

Reply:
It's optional for anonymous functions. When it's missing, the parser just assigns an empty string to `name` (which weirded me out a bit). I'd say we could either change `name` to `Option<EcoString>`, or maybe keep `name_location` as a `SrcSpan` but assigning `0` when there's no name. (Regardless, for top-level functions, they won't ever be anonymous, so we can't hit that case.)

Reply:
Cool! That makes sense, thank you.

It would be good to also add an example of an anon function in the doc comment for the struct and add a doc comment to the field saying it's optional for anonymous functions

Reply:
Done! I kept the fields there and documented the two cases (anonymous and named functions).